src/Parser/DemonHunter/Havoc/Modules/Spells/Nemesis.js

```diff
@@ -7,29 +7,66 @@ import Enemies from 'Parser/Core/Modules/Enemies';
 import SPELLS from 'common/SPELLS';
 import SpellIcon from 'common/SpellIcon';
 
-import { formatPercentage, formatDuration } from 'common/format';
+import calculateEffectiveDamage from 'Parser/Core/calculateEffectiveDamage';
+import { formatNumber, formatPercentage } from 'common/format';
 import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
 
+const NEMESIS_BUFF_IDS = [
+  SPELLS.NEMESIS_DEMON.id,
+  SPELLS.NEMESIS_HUMANOID.id,
+  SPELLS.NEMESIS_ABERRATION.id,
+  SPELLS.NEMESIS_BEAST.id,
+  SPELLS.NEMESIS_CRITTER.id,
+  SPELLS.NEMESIS_DRAGONKIN.id,
+  SPELLS.NEMESIS_ELEMENTAL.id,
+  SPELLS.NEMESIS_GIANT.id,
+  SPELLS.NEMESIS_MECHANICAL.id,
+  SPELLS.NEMESIS_UNDEAD.id,
+];
+
+const NEMESIS_DAMAGE_MODIFIER = 0.25;
+
 class Nemesis extends Analyzer {
   static dependencies = {
     combatants: Combatants,
     enemies: Enemies,
   };
 
+  bonusDmg = 0;
+
   on_initialized() {
     this.active = this.combatants.selected.hasTalent(SPELLS.NEMESIS_TALENT.id);
   }
 
-  statistic() {
-    const nemesisUptime = this.enemies.getBuffUptime(SPELLS.NEMESIS_TALENT.id);
-    const nemesisUptimePercentage = nemesisUptime / this.owner.fightDuration;
+  get hasNemesisBuff() {
+    const buffs = this.combatants.selected.activeBuffs();
+    return buffs.some(buff => NEMESIS_BUFF_IDS.includes(buff.ability.guid));
+  }
 
+  on_byPlayer_damage(event) {
+    if(event.targetIsFriendly) {
+      return;
```

Comment:
What's with the friendly check? What's the situation where this is needed?

Reply:
friendly fire

Reply:
![image](https://user-images.githubusercontent.com/4565223/35812267-c033f8d4-0a90-11e8-8622-c34c4035a326.png)

Reply:
Oh so like "Aura of Sacrifice can cause any damage to FF, so it needs to be checked for everywhere"?

Reply:
if you're not checking for a specific spell id doing the damage event then yes